diff --git a/arch/arm/boot/dts/sun8i-h3-nanopi-neo-air.dts b/arch/arm/boot/dts/sun8i-h3-nanopi-neo-air.dts
index 488793ae124a..bcd49d3e4a4b 100644
--- a/arch/arm/boot/dts/sun8i-h3-nanopi-neo-air.dts
+++ b/arch/arm/boot/dts/sun8i-h3-nanopi-neo-air.dts
@@ -85,6 +85,10 @@
 		pinctrl-0 = <&wifi_en_nanopi_neo_air>;
 		reset-gpios = <&r_pio 0 7 GPIO_ACTIVE_LOW>; /* PL7 */
 	};
+        power_ctrl: power_ctrl {
+                compatible = "gpio-poweroff";
+                gpios = <&pio 0 6 GPIO_ACTIVE_HIGH>; /* PA6 */
+        };
 
         vdd_cpux: gpio-regulator {
                 compatible = "regulator-gpio";
