diff --git a/arch/arm/boot/dts/sun8i-h3-nanopi-neo-air.dts b/arch/arm/boot/dts/sun8i-h3-nanopi-neo-air.dts
index b3a528f59075..6393c0ba5960 100644
--- a/arch/arm/boot/dts/sun8i-h3-nanopi-neo-air.dts
+++ b/arch/arm/boot/dts/sun8i-h3-nanopi-neo-air.dts
@@ -273,22 +273,38 @@
         status = "okay";
 };
 
-&ehci3 {
+&ehci0 {
 	status = "okay";
 };
 
+&ehci1 {
+        status = "okay";
+};
+
 &ehci2 {
 	status = "okay";
 };
 
-&ohci3 {
+&ehci3 {
+        status = "okay";
+};
+
+&ohci0 {
 	status = "okay";
 };
 
+&ohci1 {
+        status = "okay";
+};
+
 &ohci2 {
 	status = "okay";
 };
 
+&ohci3 {
+        status = "okay";
+};
+
 &usb_otg {
 	dr_mode = "otg";
 	status = "okay";
